packages/babel-plugin/src/ensureReactImport.ts

```diff
@@ -2,29 +2,21 @@ import { NodePath } from "@babel/core";
 import { types } from "@babel/core";
 
 /**
- * Ensures that a React import declaration exists in the given program.
- * If not, it adds the import declaration at the beginning of the program.
+ * Ensures that React is imported in the given program to satisfy JSX's Classic runtime requirement.
+ * This function adds a React import declaration at the beginning of the program with a unique alias to avoid potential naming conflicts.
  *
- * @param {NodePath<t.Program>} programPath - The path to the Program node in the AST.
- * @param {typeof t} t - The Babel types.
+ * Note: The unique alias "__React_Kuma_" is utilized in Kuma's compiler to safely refer to React (e.g., when using React.forwardRef to wrap styled components) without interfering with user-defined variable names.
+ *
+ * @param {NodePath<types.Program>} programPath - The path to the Program node in the AST.
+ * @param {typeof types} t - The Babel types.
  */
 export function ensureReactImport(
   programPath: NodePath<types.Program>,
   t: typeof types,
 ) {
-  let hasReactImport = false;
-
-  programPath.node.body.forEach((node) => {
-    if (t.isImportDeclaration(node) && node.source.value === "react") {
-      hasReactImport = true;
-    }
-  });
-
-  if (!hasReactImport) {
-    const reactImportDeclaration = t.importDeclaration(
-      [t.importDefaultSpecifier(t.identifier("React"))],
-      t.stringLiteral("react"),
-    );
-    programPath.unshiftContainer("body", reactImportDeclaration);
-  }
+  const reactImportDeclaration = t.importDeclaration(
+    [t.importDefaultSpecifier(t.identifier("__React_Kuma_"))],
+    t.stringLiteral("react"),
+  );
+  programPath.unshiftContainer("body", reactImportDeclaration);
```

Comment:
📝 (Just a memo of my understanding) We already ensure that React is imported here, but we can't determine the specifier name if default import exists, therefore we need to insert another import for Kuma.

Reply:
Thanks for the clarification. I've incorporated your comment into the JSDoc.